ivy/functional/ivy/nest.py

```diff
@@ -1055,6 +1055,61 @@ def nested_map(
     ret
         x following the applicable of fn to it's nested leaves, or x itself if x is not
         nested.
+
+    Examples
+    --------
+    With :class:`Tuple` inputs:
+
+    >>> x = ([[1., 2.], [3., 4.]])
+    >>> function = lambda a : a * 2
+    >>> ivy.nested_map(x, function)
+    >>> print(x)
+    ([[2.0, 4.0], [6.0, 8.0]])
+
+    With :code:`Dict` input:
+
+    >>> x = {1 : [1, [2, 3]], 2: (4, 5)}
+    >>> function = lambda a : a + 1
+    >>> ivy.nested_map(x, function)
+    >>> print(x)
+    {1 : [2, [3, 4]], 2: (5, 6)}
+
+    With :code:`List` inputs:
+
+    >>> x = [['a', 'b', 'c'],
+    ...      ['d', 'e', 'f'],
+    ...      ['g', ['h', 'i']]]
+    >>> function = lambda a: a + 'H'
+    >>> ivy.nested_map(x, function)
+    >>> print(x)
+    [['aH','bH','cH'],['dH','eH','fH'],['gH',['hH','iH']]]
+
+    With :class:`ivy.Container` input:
+
+    >>> x = ivy.Container(
+      a=ivy.array([[1, 2, 3], [9, 8, 7]]) , b=ivy.array([[4, 5, 6], [12, 13, 14]])
+    )
+    >>> function = lambda a : a  + 1
+    >>> ivy.nested_map(x, function)
+    >>> print(x)
+    {
+       a: ivy.array([[1, 2, 3], [9, 8, 7]]),
+       b: ivy.array([[4, 5,  6], [12, 13, 14]])
+    }
+
+    >>> nest = [[1, 2], [3, 4], [5, 6]]
+    >>> ivy.nested_map(lambda x: x * 2, nest, to_ignore=(list, dict))
+    [[1, 2], [3, 4], [5, 6]]
+
+    >>> nest = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]]
+    >>> ivy.nested_map(lambda x: x * 2, nest, max_depth=1)
+    [[[2, 4], [6, 8]], [[10, 12], [14, 16]]]
```

Comment:
Again, I can't see what `max_depth` does as it seems like it applies the lambda function to the whole object. Can you experiment with that parameter a little bit to make show an example that the lambda function is partially applied somehow. Of course it should be an actual valid code with its output